cloudfoundry-client-lib/src/main/java/org/cloudfoundry/client/lib/rest/CloudControllerRestClientImpl.java

```diff
@@ -1397,6 +1403,37 @@ public void bindDropletToApp(UUID dropletGuid, UUID applicationGuid) {
                 .block();
     }
 
+    @Override
+    public DropletInfo getCurrentDropletForApplication(UUID applicationGuid) {
+        GetApplicationCurrentDropletResponse getApplicationCurrentDropletResponse = delegate.applicationsV3()
+                                                                                            .getCurrentDroplet(GetApplicationCurrentDropletRequest.builder()
+                                                                                                                                                  .applicationId(applicationGuid.toString())
+                                                                                                                                                  .build())
+                                                                                            .block();
+        String packageUrl = getApplicationCurrentDropletResponse.getLinks()
+                                                                .get("package")
+                                                                .getHref();
+        String packageGuid = packageUrl.substring(packageUrl.lastIndexOf("/") + 1);
```

Comment:
Is this mechanism fail-safe?

Reply:
A droplet is always bound to a package. A package could have more than one droplet so I assume it is safe. The response should always have link to the package: https://v3-apidocs.cloudfoundry.org/version/3.86.0/index.html#get-current-droplet